### ISSUE TITLE
Documentation source code can be tested

### ISSUE THREAD

**Issue by bmuschko** [gradle/playframework#43]
- Create test cases
- Tests pass on CI
- Potentially use [Exemplar](https://blog.gradle.org/documentation-samples-testing-exemplar)

**Comment by bmuschko** [comment]
Implementation:

- Test of build output on console: https://github.com/gradle/play/blob/master/src/docTest/groovy/com/playframework/gradle/LogOutputUserGuideSamplesIntegrationTest.groovy. I personally don't think this type of test is very useful. It just runs the `tasks` task but doesn't actually verify the output (same as in Gradle core). This test type also required custom code for Exemplar: https://github.com/gradle/exemplar/issues/38 (maybe one of the reasons the community hasn't adopted the tool for Gradle plugin development yet).
- In-depth tests: https://github.com/gradle/play/blob/master/src/docTest/groovy/com/playframework/gradle/InDepthUserGuideSamplesIntegrationTest.groovy

**Comment by big-guy** [comment]
For the samples that need additional dependencies, do you have another issue for tracking that?

**Comment by bmuschko** [comment]
@big-guy Can you elaborate? I don't quite understand what you mean.

**Comment by bmuschko** [comment]
@big-guy There's more to that test code in these samples. I think they were built with an older version of Play and are not completely incompatible: changed APIs, different imports, missing dependencies. I spent about an hour on it but couldn't make it work. I think somebody with proper knowledge of the Play API would get the solution much faster. I created https://github.com/gradle/play/issues/74 with my findings.
